environments/jvm/src/main/java/io/fission/Server.java

```diff
@@ -29,13 +40,33 @@
 
 	private static Logger logger = Logger.getGlobal();
 
-	@RequestMapping(value = "/", method = { RequestMethod.GET, RequestMethod.POST, RequestMethod.DELETE,
-			RequestMethod.PUT })
-	ResponseEntity<Object> home(RequestEntity<?> req) {
-		if (fn == null) {
-			return ResponseEntity.badRequest().body("Container not specialized");
-		} else {
-			return ((ResponseEntity<Object>) ((Function) fn).call(req, null));
+	@RequestMapping(value = "/", method = { RequestMethod.GET, RequestMethod.POST, RequestMethod.DELETE, RequestMethod.PUT })
+	ResponseEntity<Object> home(HttpServletRequest request) {
+
+		String body;
+		// Converting header in multimap
+		try {
+				MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+				Enumeration headerNames = request.getHeaderNames();
+				while (headerNames.hasMoreElements()) {
+					String key = (String) headerNames.nextElement();
+					String value = request.getHeader(key);
+					headers.put(key, singletonList(value));
+				 }
+
+				//Extract the body from the request
+				body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
```

Comment:
Is it necessary to have `System.lineSeparator()`, wouldn't it change the format of payload?
(I'm not so familiar with Java, please correct me if I'm wrong.)